### DISCUSSION TITLE
GitHub's camo tool breaks images of external hosts in READ.ME

### DISCUSSION THREAD

**Discussion by KawaMood** [community/community#140885] [Programming Help] [answered]
### Select Topic Area

Question

### Body

Hi,

_I don't know where to exactly ask for this, there are plenty of categories. Sorry if I made a mistake!_
I have a repo in which I attached an image from my own host in the `READ.ME` file. GitHub converted its URL into a `camo.githubusercontent.com` one: https://github.com/KawaMood/creative_dimension/tree/for_1.20_to_1.20.2

Problem is that the image is broken, it sounds like their system only downloaded a little part of it and save it like that. I read that camo caches images somewhere and this often happen. Is there a way to counter that?

Note: the original link of the image is the [following one](https://mc.kawamood.com/datapacks/PK_Creative_Dimension/medias/portal_structure.png). Clicking on it you it you can see that the image is complete as expected.

Thank you in advance.

**Comment by VeerammalV** [comment, marked as answer]
It sounds like the issue you're experiencing is related to GitHub's Camo image proxy, which caches and serves external images to ensure they're loaded securely over HTTPS. When the image appears broken, it's often due to a caching issue or incomplete download on their end.

Here are a few steps you can try to counter this:

1. Clear the cache: You could try updating the URL slightly by adding a query parameter like `?v=1` at the end of the image URL. This forces GitHub to re-fetch the image from your host.

2. Check the image size: Ensure that the image is not too large or excessively slow to load from your host, as this could result in timeouts during the cache process.

3. Re-upload the image: If the issue persists, you can host the image directly within your repository (e.g., in an `assets` folder) and link to it using a relative path in your `README.md` file.

Hopefully, one of these solutions resolves the issue for you!

**Reply by KawaMood** [reply]
Oh indeed the first point is a smart trick. It worked, thanks!
